src/modules/launcher/Wox.Plugin/Common/VirtualDesktopHelper.cs

```diff
@@ -0,0 +1,411 @@
+        /// </summary>
+        /// <returns>List of desktops or an empty list on failure.</returns>
+        public List<Guid> GetDesktopList()
+        {
+            if (_desktopListAutoUpdate)
+            {
+                UpdateDesktopList();
+            }
+
+            return availableDesktops;
+        }
+
+        /// <summary>
+        /// Returns the count of existing desktops
+        /// </summary>
+        /// <returns>Number of existing desktops or zero on failure.</returns>
+        public int GetDesktopCount()
+        {
+            if (_desktopListAutoUpdate)
+            {
+                UpdateDesktopList();
+            }
+
+            return availableDesktops.Count;
+        }
+
+        /// <summary>
+        /// Returns the id of the generic desktop (view) "All Desktops".
+        /// (At the moment we can't implement this because the necessary com interfaces are private.)
+        /// </summary>
+        /// <exception cref="NotImplementedException">Because this mehtod isn't implemented.</exception>
+        private void GetAllDesktopsId()
+        {
+            // Not available via public com interfaces.
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Returns the id of the desktop taht is currently visible to the user.
+        /// </summary>
+        /// <returns>Guid of the current desktop or an empty guid on failure.</returns>
+        public Guid GetCurrentDesktopId()
+        {
+            if (_desktopListAutoUpdate)
+            {
+                UpdateDesktopList();
+            }
+
+            return currentDesktop;
+        }
+
+        /// <summary>
+        /// Returns the number (position) of a desktop.
+        /// </summary>
+        /// <returns>Number of the desktop, if found, otherwise -1.</returns>
+        /// <param name="desktop">The guid of the desktop.</param>
+        public int GetDesktopNumber(Guid desktop)
+        {
+            if (_desktopListAutoUpdate)
+            {
+                UpdateDesktopList();
+            }
+
+            return availableDesktops.IndexOf(desktop);
+        }
+
+        /// <summary>
+        /// Returns the name of a desktop
+        /// </summary>
+        /// <param name="desktop">Guid of the desktop</param>
+        /// <returns>Returns the name of the desktop, or "All Desktops" if guid belongs to the generic desktop "All Desktops", or an empty string on failur.</returns>
+        public string GetNameOfDesktop(Guid desktop)
+        {
+            if (desktop == Guid.Empty)
+            {
+                return string.Empty;
+            }
+
+            // Here we expect that the guid in the parameter is valid.
+            // So if we can't find the id in the desktop list, it must be the generic id for "All Desktops".
+            if (!GetDesktopList().Contains(desktop))
+            {
+                return Properties.Resources.VirtualDesktopHelper_AllDesktops;
```

Comment:
Change to `return String.empty;`

Expecting a valid guid isn't a good way to go. And `All desktops` isn't the name of a really existing desktop.